opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -260,5 +270,27 @@ def create_metric(
             monotonic=monotonic,
         )
 
+    def get_label_set(self, labels: Dict[str, str]):
+        """See `opentelemetry.metrics.Meter.create_metric`.
+
+        This implementation encodes the labels to use as a map key.
+
+        Args:
+            labels: The dictionary of label keys to label values.
+        """
+        if len(labels) == 0:
+            return EMPTY_LABEL_SET
+        sorted_labels = OrderedDict(sorted(labels.items()))
+        # Uses statsd encoding for labels
+        encoded = "|#" + ",".join(
+            "%s:%s" % (key, value) for (key, value) in sorted_labels.items()
+        )
+        # If LabelSet exists for this meter in memory, use existing one
+        if not self.labels.get(encoded):
```

Comment:
I think this code would be better in LabelSet itself. The encoded field could be implemented as a property which lazy-populates the value in the same way that this is done.

Reply:
Wouldn't that mean the `LabelSet` instances would need to be able to access the cache of label encodings `labels` unique per `Meter` instance somehow? Would that really be cleaner?

Reply:
I agree that a lazy `LabelSet.encoded` wouldn't work since the point is to prevent instantiating the `LabelSet`, but the encoding code shouldn't be here either.